SYCL/ESIMD/esimd_test_utils.hpp

```diff
@@ -461,4 +462,28 @@ inline void iterate_ops(OpSeq<OpClass, Ops...> ops, F f) {
   ConstexprForLoop<0, sizeof...(Ops)>::unroll(act);
 }
 
+/* Randomized input generator */
+namespace input_gen {
+template <typename T> class Generator {
+public:
+  explicit Generator() : _gen(std::random_device{}()), _dist() {}
+  Generator(T ub) : _gen(std::random_device{}()), _dist(0, ub) {}
+  Generator(T lb, T ub) : _gen(std::random_device{}()), _dist(lb, ub) {}
+
+  T operator()() { return static_cast<T>(_dist(_gen)); }
+
+private:
+  using dist_t = typename std::conditional<
+      std::is_integral<T>::value,
+      typename std::conditional<std::is_same<T, char>::value ||
```

Comment:
Isn't this line redundant?: char is either unsigned or signed

Reply:
Question: if this class is supposed to be used  just with chars, why this check attached to distribution type definition, not the enclosing class?

Reply:
I did not noticed while I was importing this code from our internal validation codes. Will update.